### DISCUSSION TITLE
Most idiomatic way of getting multiple inputs

### DISCUSSION THREAD

**Discussion by matthewturk** [chhoumann/quickadd#190] [Q&A]
Hi!  I'm working on trying to capture little snippets of information to different categories chosen from a particular list of items.  So for instance, "Topic A", "Topic B" etc.  I'd like to append each of these to my daily note with a time/date stamp, and also to use metadata tagging for use with dataview.  Basically, what this means is that I want to have two inputs -- one for the category (which could be a suggester, but I'm fine with free text if need be) and one for the content.

What I'm not sure about is the "right" way to do this.  I *think* I can see a few different possibilities, but as I look at them, it feels like I'm not quite getting the best way to do it using QuickAdd.

 1. Add an individual capture for each category, with identical settings for where to place the text but different metadata, and put all of these under a multi-choice.
 2. Write a user-defined function that provides a suggester (and input) and then returns a JSON object, which I would then access via `{{VALUE.category}}` and `{{VALUE.content}}` or something similar, and have a step in the macro being to call the capture.  Call the macro directly.
 3. Evaluate a `{{MACRO}}` in a capture.

I'm still getting myself accustomed to quickadd, but I couldn't find any explicit examples of how to do this.  Does anyone have any suggestions?

**Comment by chhoumann** [comment]
Hey @matthewturk!

There are a few ways to do this. The ones you've mentioned are very valid ways.
I'd say that your approach depends on your use-case.
For example, you can spawn a suggester with `{{VALUE:A,B,C}}`, which opens a suggester like this one:
![image](https://user-images.githubusercontent.com/29108628/152507440-c1803e05-25ba-4373-9278-99f1185921fd.png)

But the catch is that you can't access the chosen value again. This is where you'd instead use some script to open a suggester and save it to a variable, which you could access like `{{VALUE:category}}`.

**Reply by Gaaruda** [reply]
@chhoumann I want to acess the names of the files in particular folder as option, please help

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that.

This has a clean, zero-script answer now. QuickAdd gained named suggesters in 2.14.0, so the catch from my 2022 reply (not being able to reuse the chosen value) is gone.

One Capture choice, capturing to your daily note, with a format like:

```
- {{TIME}} [category:: {{VALUE:Topic A,Topic B,Topic C|name:category}}] {{VALUE:content}}
```

- The list token opens a suggester, and `|name:category` makes the pick reusable: `{{VALUE:category}}` can appear again anywhere - metadata, tags, even the file path - without prompting a second time.
- `{{VALUE:content}}` is a separate free-text prompt, also prompted exactly once.
- `|label:` adds helper text to a prompt, and `|text:` lets you show friendlier display labels than the stored values, e.g. `{{VALUE:topic-a,topic-b|name:category|text:Topic A,Topic B}}`.

One placement note: if you also want `{{VALUE:category}}` in the file name/path, define the `|name:` token in the field that resolves first (the file name resolves before the capture body).

@Gaaruda, your follow-up exists natively now too: `{{FILE:<folder>}}` opens a picker over the files in that folder and inserts the picked file's name (`|link` gives you a wikilink instead).

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#value-name

Closing as resolved - the format syntax covers this without any scripting now.
